packages/react-storage/src/components/StorageBrowser/Views/Controls/AddFolder.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { withBaseElementProps } from '@aws-amplify/ui-react-core/elements';
+
+import type { OmitElements } from '../types';
+import { StorageBrowserElements } from '../../context/elements';
+import { CLASS_BASE } from '../constants';
+import { useControl } from '../../context/controls';
+
+const { Button: ButtonElement } = StorageBrowserElements;
+
+const BLOCK_NAME = `${CLASS_BASE}__add-folder`;
+
+export interface _AddFolderControl<
+  T extends StorageBrowserElements = StorageBrowserElements,
+> {
+  (): React.JSX.Element;
+  Button: T['Button'];
+}
+
+export interface AddFolderControl<
+  T extends StorageBrowserElements = StorageBrowserElements,
+> extends OmitElements<_AddFolderControl<T>, 'Button'> {
+  (): React.JSX.Element;
+}
+
+const Button = withBaseElementProps(ButtonElement, {
+  className: `${BLOCK_NAME}`,
+  variant: 'primary',
+});
+
+export const AddFolderControl: AddFolderControl = () => {
```

Comment:
Looks like we are doing a lot of duplication between the components in this PR, anything we can move in to shared utils or a hook?